src/main/java/org/jvnet/hudson/update_center/Main.java

```diff
@@ -227,6 +229,16 @@ private JSONObject buildUpdateCenterJson(MavenRepository repo, LatestLinkBuilder
         return root;
     }
 
+    private JSONArray buildWarnings() {
+        try {
+            String warningsText = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream("warnings.json"));
+            JSONArray warnings = JSONArray.fromObject(warningsText);
```

Comment:
Uh… why would it ever not exist? Also, remember that this is a periodically run tool. If it doesn't exist, I'd hope it'd abort with error and fail the generation build.